src/include/iceberg_metadata.hpp

```diff
@@ -23,20 +55,28 @@ class IcebergSnapshot {
 	string manifest_list;
 	timestamp_t timestamp_ms;
 	idx_t iceberg_format_version;
+	uint64_t schema_id;
+	vector<IcebergColumnDefinition> schema;
 
 	static IcebergSnapshot GetLatestSnapshot(string &path, FileSystem &fs);
 	static IcebergSnapshot GetSnapshotById(string &path, FileSystem &fs, idx_t snapshot_id);
 	static IcebergSnapshot GetSnapshotByTimestamp(string &path, FileSystem &fs, timestamp_t timestamp);
 
-	static IcebergSnapshot ParseSnapShot(yyjson_val *snapshot, idx_t iceberg_format_version);
+	static IcebergSnapshot ParseSnapShot(yyjson_val *snapshot, idx_t iceberg_format_version, idx_t schema_id,
+	                                     vector<yyjson_val *> &schemas);
 	static string ReadMetaData(string &path, FileSystem &fs);
+	static yyjson_val *GetSnapshots(string &path, FileSystem &fs);
+	//! Note: caller keeps ownership of yyjson doc
+	static unique_ptr<SnapshotParseInfo> GetParseInfo(yyjson_doc *metadata_json);
```

Comment:
Maybe this should take a reference instead of a raw pointer, that might better convey that this function doesn't take ownership over the yyjson doc

Reply:
good one, that seems nicer